### PR TITLE
[JUJU-1122] Reject if --build-agent was provided for a snap juju;

### PR DESCRIPTION
**Pull request by ycliuhw**
`--build-agent` is purely for local development purposes, so disallow to use it for snap juju client.

## Checklist

 - [ ] ~Requires a [pylibjuju](https://github.com/juju/python-libjuju) change~
 - [ ] ~Added [integration tests](https://github.com/juju/juju/tree/develop/tests) for the PR~
 - [ ] ~Added or updated [doc.go](https://discourse.charmhub.io/t/readme-in-packages/451) related to packages changed~
 - [x] Comments answer the question of why design decisions were made

## QA steps

```console
$ which juju
/snap/bin/juju

$ juju bootstrap lxd k1 --build-agent
Creating Juju controller "k1" on lxd/localhost
Building local Juju agent binary version 2.9.31 for amd64
ERROR failed to bootstrap model: cannot package bootstrap agent binary: can not build agent for official build

```

## Documentation changes

No

## Bug reference

https://bugs.launchpad.net/juju/+bug/1812191

### CODE REVIEW

File: environs/tools/build.go [hunk 1]
```diff
@@ -301,23 +301,43 @@ func packageLocalTools(toolsDir string, buildAgent bool) error {
...
 	dir, err := ioutil.TempDir("", "juju-tools")
 	if err != nil {
 		return version.Binary{}, false, "", err
 	}
 	defer os.RemoveAll(dir)
+
+	existingJujuLocation, err := ExistingJujuLocation()
+	if err != nil {
+		return version.Binary{}, false, "", errors.Annotate(err, "couldn't find existing jujud")
+	}
```
Comment (wallyworld) [inline environs/tools/build.go:326]: If build is specified and there's no existing jujud, we don't want to error

Comment (ycliuhw) [reply environs/tools/build.go:326]: the location here is the juju client but not jujud

**Review by wallyworld — APPROVED** [review]
I think this is an improvement for sure

File: environs/tools/build.go [hunk 2]
```diff
@@ -301,23 +301,43 @@ func packageLocalTools(toolsDir string, buildAgent bool) error {
...
+	_, official, err = jujudVersion(existingJujuLocation)
+	if err != nil {
+		return version.Binary{}, official, "", errors.Trace(err)
+	}
+	if official && build {
+		return version.Binary{}, official, "", errors.Errorf("can not build agent for official build")
```
Comment (barrettj12) [inline environs/tools/build.go:332]: `cannot build agent` would be more grammatical

**Pull request merged**